syne_tune/optimizer/schedulers/searchers/kde_searcher.py

```diff
@@ -0,0 +1,275 @@
+import syne_tune.search_space as sp
+
+__all__ = ['KernelDensityEstimator']
+
+logger = logging.getLogger(__name__)
+
+
+class KernelDensityEstimator(BaseSearcher):
+    """
+    Fits two kernel density estimators (KDE) to model the density of the top N configurations as well as the density
+    of the configurations that are not among the top N, respectively. New configurations are sampled by optimizing
+    the ratio of these two densities. KDE as model for Bayesian optimization has been originally proposed
+    by Bergstra et al. Compared to their original implementation TPE, we use multi-variate instead of univariate KDE
+    as proposed by Falkner et al.
+    Code is based on the implementation by Falkner et al: https://github.com/automl/HpBandSter/tree/master/hpbandster
+
+    Algorithms for Hyper-Parameter Optimization
+    J. Bergstra and R. Bardenet and Y. Bengio and B. K{\'e}gl
+    Proceedings of the 24th International Conference on Advances in Neural Information Processing Systems
+
+    BOHB: Robust and Efficient Hyperparameter Optimization at Scale
+    S. Falkner and A. Klein and F. Hutter
+    Proceedings of the 35th International Conference on Machine Learning
+
+
+    Parameters
+    ----------
+    config_space: dict
+        Configuration space for trial evaluation function
+    metric : str
+        Name of metric to optimize, key in result's obtained via
+        `on_trial_result`
+    mode : str
+        Mode to use for the metric given, can be 'min' or 'max', default to 'min'.
+    num_min_data_points: int
+        Minimum number of data points that we use to fit the KDEs. If set to None than we set this to the number of
+        hyperparameters.
+    top_n_percent: int
+        Determines how many datapoints we use use to fit the first KDE model for modeling the well
+        performing configurations.
+    min_bandwidth: float
+        The minimum bandwidth for the KDE models
+    num_candidates: int
+        Number of candidates that are sampled to optimize the acquisition function
+    bandwidth_factor: int
+        We sample continuous hyperparameter from a truncated Normal. This factor is multiplied to the bandwidth to
+        define the standard deviation of this trunacted Normal.
+    random_fraction: float
+        Defines the fraction of configurations that are drawn uniformly at random instead of sampling from the model
+    points_to_evaluate: List[Dict] or None
+        List of configurations to be evaluated initially (in that order).
+        Each config in the list can be partially specified, or even be an
+        empty dict. For each hyperparameter not specified, the default value
+        is determined using a midpoint heuristic.
+        If None (default), this is mapped to [dict()], a single default config
+        determined by the midpoint heuristic. If [] (empty list), no initial
+        configurations are specified.
+    """
+
+    def __init__(
```

Comment:
I suppose this searcher does not work together with multi-fidelity schedulers, right?

Could you implement configure_scheduler as in gp_fifo_searcher, checking that the scheduler is FIFOScheduler? The error message would tell people to use MultiFidelityKernelDensityEstimator instead.